File: app/src/main/java/com/fullmeadalchemist/mustwatch/vo/Group.java
package com.fullmeadalchemist.mustwatch.vo;

import android.arch.persistence.room.Entity;
import android.arch.persistence.room.Index;
import android.arch.persistence.room.PrimaryKey;


@Entity(tableName = "groups",
        indices = {@Index(value = {"name"}, unique = true)})
public class Group {

    @PrimaryKey(autoGenerate = true)
    public long id;
    public String name;

    public Group(String name) {
        this.name = name;
    }
}

File: app/src/main/java/com/fullmeadalchemist/mustwatch/vo/Group.kt
/*
 * Copyright (c) 2017 Full Mead Alchemist, LLC.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.fullmeadalchemist.mustwatch.vo

import android.arch.persistence.room.ColumnInfo
import android.arch.persistence.room.Entity
import android.arch.persistence.room.Index
import android.arch.persistence.room.PrimaryKey
import android.support.annotation.NonNull

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName


@Entity(tableName = "groups", indices = [Index(value = ["name"], unique = true)])
data class Group(@field:Expose @field:SerializedName("name") @field:ColumnInfo(name = "name") var name: String) {

    @Expose
    @NonNull
    @PrimaryKey(autoGenerate = true)
    @SerializedName("id")
    @ColumnInfo(name = "id")
    var id: Long = 0
}
